### PR TITLE
Update deprecated `QNode.tape` property usage in `tests`

### PR DESCRIPTION
**Pull request by andrijapau**
**Context:**

Follow up to #6625.

For some reason, CI never complained about this property usage despite no `qml.PennylaneDeprecationWarning` suppression occuring in these files ... weird. Created story ([sc-80957]) to track.

**Description of the Change:**

Tests should construct the tape dynamically using `construct_tape` rather than use the deprecated properties.

**Benefits:** Better testing and code maintainability.

**Possible Drawbacks:** None

[[sc-78317](https://app.shortcut.com/xanaduai/story/78317)]

### CODE REVIEW

File: tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py [hunk 1]
```diff
@@ -261,12 +261,13 @@ def circuit(params):
             return qml.probs([2, 3])
 
         params = np.array([0.5, 0.5, 0.5], requires_grad=True)
-        circuit.construct((params,), {})
```
Comment (PietropaoloFrisoni) [inline tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py:264]: Are we removing the `construct` usage because it internally refers to the `_tape` variable, right?

File: tests/test_return_types.py [hunk 2]
```diff
@@ -40,13 +40,12 @@ def circuit(x):
...
         qnode = qml.QNode(circuit, dev)
-        qnode.construct([0.5], {})
-
+        tape = qml.workflow.construct_tape(qnode)(0.5)
         if dev.shots:
             pytest.skip("cannot return analytic measurements with finite shots.")
         program = dev.preprocess_transforms()
         res = qml.execute(
-            tapes=[qnode.tape],
+            tapes=[tape],
```
Comment (PietropaoloFrisoni) [inline tests/test_return_types.py:48]:
```suggestion
            tapes=[qml.workflow.construct_tape(qnode)(0.5)],
```

File: tests/test_return_types.py [hunk 3]
```diff
@@ -40,13 +40,12 @@ def circuit(x):
             return qml.state()
 
         qnode = qml.QNode(circuit, dev)
-        qnode.construct([0.5], {})
-
+        tape = qml.workflow.construct_tape(qnode)(0.5)
```
Comment (PietropaoloFrisoni) [inline tests/test_return_types.py:43]:
```suggestion
```

File: tests/test_return_types.py (hunk 2 above)
Comment (PietropaoloFrisoni) [reply tests/test_return_types.py:48]: Just for consistency with the rest of the test file : )

**Review by PietropaoloFrisoni — APPROVED** [review]
Thanks @andrijapau :100:

File: tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py (hunk 1 above)
Comment (andrijapau) [reply tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py:264]: The `QNode` is constructed when you call `qml.workflow.construct_tape` 😄. Therefore, I'm just removing some redundant code.

**Review by lillian542 — APPROVED** [review]
Looks good to me!
